gitlab/client.py

```diff
@@ -832,15 +835,33 @@ def http_list(
         """
         query_data = query_data or {}
 
-        # In case we want to change the default behavior at some point
-        as_list = True if as_list is None else as_list
+        # Don't allow both `as_list` and `iterator` to be set.
+        if as_list is not None and iterator is not None:
+            raise gitlab.exceptions.GitlabError(
+                error_message=(
+                    "`as_list` and `iterator` are mutually exclusive. Only one "
+                    "argument can be set. Recommend to only set `iterator` argument"
+                )
+            )
+
+        if as_list is not None:
+            iterator = not as_list
+            utils.warn(
+                message=(
+                    f"`as_list={as_list}` has been deprecated. Please use "
+                    f"`iterator={iterator}` instead"
+                ),
+                category=DeprecationWarning,
+            )
+        if iterator is None:
+            iterator = False
```

Comment:
IMO in this case this is redundant - we have `if iterator:` a few lines down from here so I think it's quite idiomatic without it as there isn't any user code messing with this in between.

```suggestion
```

Reply:
Done